relational.tex

```diff
@@ -553,104 +549,136 @@ \subsection{Incremental view maintenance}
     \node[block, right of=I] (z) {$\zm$};
     \node[block, below of=z, node distance=.8cm] (H) {$\lift{H}$};
     \node[right of=H] (output) {$\Delta o$};
-    \draw[->] (input) -- node (mid) {} (I);
-    \draw[->] (I) -- (z);
-    \draw[->] (mid.center) |- (H);
-    \draw[->] (z) -- node (i) [right] {$i$} (H);
-    \draw[->] (H) -- (output);
+    \draw[->>] (input) -- node (mid) {} (I);
+    \draw[->>] (I) -- (z);
+    \draw[->>] (mid.center) |- (H);
+    \draw[->>] (z) -- node (i) [right] {$i$} (H);
+    \draw[->>] (H) -- (output);
 \end{tikzpicture}
 \end{tabular}
 
-\noindent where $H: \Z[A] \times \Z[A] \to \Z[A]$ is defined as: \\
-$$
-H(i, d)[x] \defn
-\begin{cases}
--1 & \mbox{if } i[x] > 0 \mbox{ and } (i + d)[x] \leq 0 \\
-1  & \mbox{if } i[x] \leq 0 \mbox{ and } (i + d)[x] > 0 \\
-0  & \mbox{otherwise} \\
-\end{cases}
-$$
-\end{proposition}
+%\noindent where $H: \Z[A] \times \Z[A] \to \Z[A]$ is defined as: \\
+%$$
+%H(i, d)[x] \defn
+%\begin{cases}
+%-1 & \mbox{if } i[x] > 0 \mbox{ and } (i + d)[x] \leq 0 \\
+%1  & \mbox{if } i[x] \leq 0 \mbox{ and } (i + d)[x] > 0 \\
+%0  & \mbox{otherwise} \\
+%\end{cases}
+%$$
+%\end{proposition}
 
 Here is the intuition why $\distinct$ is efficiently
-incrementalizable: the only elements that can appear in the output of
-$\inc{(\lift{\distinct})}$ must have changed in the input.  So the
-size of the output change cannot be bigger than the size of the input
-change.  In the diagram above, $i$ is the previous version of the
-integral of all changes, i.e., the full \zr whose $\distinct$ value is
-being computed.  The function $H$ detects whether the weight of
-an element in $i$ is changing sign (positive to negative or
-vice-versa) when adding a new delta $d$.
+incrementalizable: consider a row that is inserted in the input of a
+$\distinct$ operator.  How does the output change?  If the row was not
+in the prior full input table, then it will appear with weight 1 in
+the output change.  Otherwise an empty output is produced.  Only
+tuples that appear in the input change $\Delta d$ can appear in the
+output change $\Delta o$.  The implementation needs to maintain the
```

Comment:
maybe clarify this point a bit more: 

Only tuples that appear in the input change $\Delta d$ can appear in the output change $\Delta o$ and hence the operator only needs to perform $|Delta d|$ lookups.